src/components/v2/appDetails/index.store.ts

```diff
@@ -0,0 +1,259 @@
+        _nodes.forEach((_n: Node) => {
+            _n.parentRefs?.forEach((_pr) => {
+                if (_pr.uid === _pn.uid) {
+                    childNodes.push(_n as iNode);
+                }
+            });
+        });
+
+        if (childNodes.length > 0) {
+            fillChildNodes(childNodes, _nodes, _kind);
+
+            _pn.childNodes = childNodes;
+        }
+
+        if (_kind?.toLowerCase() === NodeType.Pod.toLowerCase()) {
+            _pn.childNodes = _appDetailsSubject
+                .getValue()
+                .resourceTree?.podMetadata?.filter((_pmd) => {
+                    return _pmd.uid === _pn.uid;
+                })[0]
+                ?.containers?.map((_c) => {
+                    let childNode = {} as iNode;
+                    childNode.kind = NodeType.Containers;
+                    childNode['pNode'] = _pn;
+                    childNode.name = _c;
+                    return childNode;
+                });
+        }
+
+        return _pn;
+    });
+};
+
+const getAllParentNods = (_nodes: Array<Node>, _kind: string): Array<iNode> => {
+    let _allParentNodes = [];
+    let _allParentNodeTypes = [];
+
+    _nodes.forEach((_n) => {
+        _n.parentRefs?.forEach((_prn: Node) => {
+            let prn = _n as iNode;
+            _allParentNodes.push(prn);
+            _allParentNodeTypes.push(_prn.kind);
+        });
+    });
+
+    return fillChildNodes(_allParentNodes, _nodes, _kind);
+};
+
+const IndexStore = {
+    setEnvDetails: (envType: string, appId: number, envId: number) => {
+        let _envDetails = {} as EnvDetails;
+
+        _envDetails.envType = envType as EnvType;
+        _envDetails.appId = appId;
+        _envDetails.envId = envId;
+
+        _nodeFilter = {
+            filterType: '',
+            searchString: '',
+        };
+
+        _envDetailsSubject.next({ ..._envDetails });
+    },
+
+    getEnvDetails: () => {
+        return _envDetailsSubject.getValue();
+    },
+
+    getEnvDetailsObservable: () => {
+        return _envDetailsSubject.asObservable();
+    },
+
+    setAppDetails: (data: AppDetails) => {
+        console.log('setAppDetails', data);
+
+        const _nodes = data.resourceTree.nodes;
+
+        _appDetailsSubject.next({ ...data });
+
+        _nodesSubject.next([..._nodes]);
+
+        _nodesFilteredSubject.next([..._nodes]);
+    },
+
+    getAppDetails: () => {
+        return _appDetailsSubject.getValue();
+    },
+
+    getAppDetailsObservable: () => {
+        return _appDetailsSubject.asObservable();
+    },
+
+    getAppDetailsNodes: () => {
+        return _nodesSubject.getValue();
+    },
+
+    getAppDetailsNodesObservable: () => {
+        return _nodesSubject.asObservable();
+    },
+
+    getAppDetailsFilteredNodes: () => {
+        return _nodesFilteredSubject.getValue();
+    },
+
+    getAppDetailsNodesFilteredObservable: () => {
+        return _nodesFilteredSubject.asObservable();
+    },
+
+    getiNodesByKind: (_kind: string) => {
```

Comment:
dont understand the puropose of this function. How is it different from getNodesByKind

Reply:
We are using it in resource tree node